pyiron/atomistics/structure/atoms.py

```diff
@@ -1118,24 +1118,75 @@ def get_extended_positions(self, width):
         v_repeated = v_repeated[check_dist]
         return v_repeated, indices
 
+    def get_number_of_neighbors_in_sphere(
+            self,
+            cutoff_radius=10,
+            num_neighbors=None,
+            tolerance=2,
+            id_list=None,
+            boundary_width_factor=1.2,
+    ):
+        """
+        Function to compute the number of neighbors in a sphere around each atom.
+        Args:
+            cutoff_radius (float): Upper bound of the distance to which the search must be done
+            num_neighbors (int): maximum number of neighbors found
+            tolerance (int): tolerance (round decimal points) used for computing neighbor shells
+            id_list (list): list of atoms the neighbors are to be looked for
+            boundary_width_factor (float): width of the layer to be added to account for pbc.
+
+        Returns:
+            maximum number of neigbors found in the sphere of radius cutoff_radius (<= num_neighbors if specified)
+        """
+        if num_neighbors is not None:
+            neigh = self._get_neighbors(
+                num_neighbors=num_neighbors,
+                t_vec=False,
+                tolerance=tolerance,
+                id_list=id_list,
+                cutoff_radius=cutoff_radius,
+                boundary_width_factor=boundary_width_factor,
+            )
+            num_neighbors = max(len(elem) for elem in neigh.indices)
+        else:
+            num_neighbors = int(0.5*cutoff_radius**3)
+            num_neighbors_old = 1
+            while num_neighbors_old < num_neighbors:
+                neigh = self._get_neighbors(
+                    num_neighbors=num_neighbors,
+                    t_vec=False,
+                    tolerance=tolerance,
+                    id_list=id_list,
+                    cutoff_radius=cutoff_radius,
+                    boundary_width_factor=boundary_width_factor,
+                )
+                neigh.indices = [indices[dist < np.inf] for indices, dist in zip(neigh.indices, neigh.distances)]
```

Comment:
Do I see it correctly that what you want is equivalent to:

```
num_neighbors = np.sum(neigh.distances<np.inf, axis=-1).max()
```

?

Reply:
Might be correct. Summing over Booleans is converted to summing 1 or 0, axis=-1 tells to sum all axes except the first one. Therefore, the result is a vector with the dimension of the inner vectors. The maximum value of the vector, indeed, is the number of neighbors I am searching. Please correct me, if I understand any of the points wrong!